editor/src/clj/editor/pipeline/texture_set_gen.clj

```diff
@@ -77,38 +76,73 @@
     :sprite-trim-mode-7 7
     :sprite-trim-mode-8 8))
 
+(defn- texture-set-layout-rect
+  ^TextureSetLayout$Rect [{:keys [path width height]}]
+  (let [id (resource/proj-path path)]
+    (TextureSetLayout$Rect. id -1 (int width) (int height))))
+
+(defonce ^:private ^TextureSetProto$SpriteGeometry rect-sprite-geometry-template
+  (let [points (vector-of :float -0.5 -0.5 -0.5 0.5 0.5 0.5 0.5 -0.5)
+        uvs (vector-of :float 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0)
+        indices (vector-of :int 0 1 2 0 2 3)]
+    (-> (TextureSetProto$SpriteGeometry/newBuilder)
+        (.addAllVertices points)
+        (.addAllUvs uvs)
+        (.addAllIndices indices)
+        (.buildPartial))))
+
+(defn- make-rect-sprite-geometry
+  ^TextureSetProto$SpriteGeometry [^long width ^long height]
+  (-> (TextureSetProto$SpriteGeometry/newBuilder rect-sprite-geometry-template)
+      (.setWidth width)
+      (.setHeight height)
+      (.build)))
+
+(defn- make-image-sprite-geometry
+  ^TextureSetProto$SpriteGeometry [^Image image]
+  (let [error-node-id (:digest-ignored/error-node-id image)
+        resource (.path image)
+        sprite-trim-mode (.sprite-trim-mode image)]
+    (assert (g/node-id? error-node-id))
+    (assert (resource/resource? resource))
+    (case sprite-trim-mode
+      :sprite-trim-mode-off (make-rect-sprite-geometry (.width image) (.height image))
+      (let [buffered-image (resource-io/with-error-translation resource error-node-id :image
+                             (image-util/read-image resource))]
+        (g/precluding-errors buffered-image
+          (let [hull-vertex-count (sprite-trim-mode->hull-vertex-count sprite-trim-mode)]
+            (TextureSetGenerator/buildConvexHull buffered-image hull-vertex-count)))))))
+
```

Comment:
With `:sprite-trim-mode-off` we still need to include `TextureSetProto$SpriteGeometry` entries in the protobuf data. Here we make rectangle geometries for each such image, or proper convex hulls for images with Sprite Trimming enabled.